hothouse/ray_callbacks.pyx

```diff
@@ -1,10 +1,113 @@
 # cython: language=c++
 
+cimport numpy as np
+import numpy as np
 from pyembree.callback_handler cimport \
     RayCollisionCallback, CALLBACK_TERMINATE, CALLBACK_CONTINUE
+from pyembree.rtcore_geometry cimport RTC_INVALID_GEOMETRY_ID
 from pyembree.rtcore_ray cimport RTCRay
 
 cdef class RayCollisionPrinter(RayCollisionCallback):
-    cdef int callback(self, RTCRay ray):
+    cdef int callback(self, RTCRay &ray):
         print("Hi!", ray.geomID)
         return CALLBACK_TERMINATE
+
+
+cdef class RayCollisionMultiBounce(RayCollisionCallback):
```

Comment:
This makes me think we should only use callbacks and use intersection queries while setting output to False when calling pyembree.